src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -203,6 +203,12 @@ internal static TarEntryType GetCorrectTypeFlagForFormat(TarEntryFormat format,
         internal static T ParseOctal<T>(ReadOnlySpan<byte> buffer) where T : struct, INumber<T>
         {
             buffer = TrimEndingNullsAndSpaces(buffer);
+            buffer = TrimLeadingNullsAndSpaces(buffer);
```

Comment:
This is fine, most tools I have tried only ignore nulls and spaces. I haven't found a spec explicitly calling it out how lenient the implementation can or should be, but libarchive's behavior is that it ignores all non-octal characters.

Maybe add a comment linking to formal specification which talks about it, or if there is none:
"The formal specification does not specify it but we only trim nulls and spaces"

Reply:
Thanks @am11. Sure, I can add a comment. If I don't get any blocking feedback to address, I'll add the comment in a separate PR to not block this with a CI re-run.